packages/adapter-axios/src/index.ts

```diff
@@ -1,36 +1,26 @@
-import { AxiosError, AxiosInstance, AxiosResponse, Method, AxiosRequestConfig } from 'axios'
+import type { IAdapter, AdapterResponse, AdapterError, AdapterOptions } from '@openapi-client/adapter-types'
+import { AxiosInstance, AxiosRequestConfig, Method } from 'axios'
 
-export interface AdapterResponse<T> extends AxiosResponse<T> {
-	status: number
-	statusText: string
-	data: T
-	headers: Record<string, string>
-}
-
-export interface AdapterError<T> extends AxiosError<T> {
-	code?: string
-	response?: AdapterResponse<T>
-	isOpenApiError: boolean
-}
+export interface AxiosAdapterOptions extends AdapterOptions, AxiosRequestConfig {}
 
-export class AxiosAdapter {
+export class AxiosAdapter implements IAdapter {
 	constructor(
 		public axios: AxiosInstance,
-		public options: AxiosRequestConfig = {},
```

Comment:
In context of this axios adapter is better to use `AxiosRequestConfig`. To keep compatibility with interface IAdapter exact type, in this case `AxiosRequestConfig` must be passed to `IAdapter`. 

Interface should be defined with template param like this:
```
export interface IAdapter<TOptions extends AdapterOptions> {
  withOptions(options: TOptions): IAdapter<TOptions>
}
```

Reply:
Thanks, now I get it and will change it 👍